src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    public class CodeGenAsyncMainTests : EmitMetadataTestBase
```

Comment:
>CodeGenAsyncMainTests [](start = 17, length = 21)

Please add tests for combination of multiple candidates, both error and success cases, if any are possible.